Mathlib/CategoryTheory/Limits/FilteredColimitCommutesFiniteLimit.lean

```diff
@@ -0,0 +1,415 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+
+! This file was ported from Lean 3 source module category_theory.limits.filtered_colimit_commutes_finite_limit
+! leanprover-community/mathlib commit 3f409bd9df181d26dd223170da7b6830ece18442
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Limits.ColimitLimit
+import Mathlib.CategoryTheory.Limits.Preserves.FunctorCategory
+import Mathlib.CategoryTheory.Limits.Preserves.Finite
+import Mathlib.CategoryTheory.Limits.Shapes.FiniteLimits
+import Mathlib.CategoryTheory.Limits.Preserves.Filtered
+import Mathlib.CategoryTheory.ConcreteCategory.Basic
+
+/-!
+# Filtered colimits commute with finite limits.
+
+We show that for a functor `F : J × K ⥤ Type v`, when `J` is finite and `K` is filtered,
+the universal morphism `colimit_limit_to_limit_colimit F` comparing the
+colimit (over `K`) of the limits (over `J`) with the limit of the colimits is an isomorphism.
+
+(In fact, to prove that it is injective only requires that `J` has finitely many objects.)
+
+## References
+* Borceux, Handbook of categorical algebra 1, Theorem 2.13.4
+* [Stacks: Filtered colimits](https://stacks.math.columbia.edu/tag/002W)
+-/
+
+
+universe v u
+
+open CategoryTheory
+
+open CategoryTheory.Category
+
+open CategoryTheory.Limits.Types
+
+open CategoryTheory.Limits.Types.FilteredColimit
+
+namespace CategoryTheory.Limits
+
+variable {J K : Type v} [SmallCategory J] [SmallCategory K]
+
+variable (F : J × K ⥤ Type v)
+
+-- **TODO** naming? placing?
+/-- `(curry.obj F ⋙ lim).obj S` = `limit ((curry.obj F).obj S)` definitionally, so this
+is just a variant of `limit_ext'`. -/
+@[ext] lemma curry_obj_ggg_lim_obj_ext (x y : (curry.obj F ⋙ lim).obj S) (w : ∀ (j : K),
+    limit.π ((curry.obj F).obj S) j x = limit.π ((curry.obj F).obj S) j y) : x = y :=
+  limit_ext' _ x y w
```

Comment:
This fixes an `ext` problem in the file. Any suggestions for names and/or location?

Reply:
The fact that `ext` won't just do this is really making me sad. :-( It is defeating the design principle that everything boring in the category theory library should be aspirationally be provable `by intros; ext; simp`.

I could just write an `ext!` that isn't a squeamish as the built-in one?

Reply:
This lemma has the wrong name. Can `curry` just be replaced with an arbitrary functor or can this be generalized in some way (I don't think so).

Reply:
I tried removing `curry` when I wrote it and I failed, but I just tried again and it worked fine. What do you think of the current name/statement @ChrisHughes24 ?

Reply:
Looks good to me!